services/frontend/domain/usecases/process_manifest_files.usecase.go

```diff
@@ -0,0 +1,154 @@
+package usecases
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/rs/zerolog/log"
+	"gopkg.in/yaml.v3"
+
+	"github.com/berops/claudie/internal/manifest"
+	"github.com/berops/claudie/proto/pb"
+)
+
+// ProcessManifestFiles processes the manifest files concurrently. If an error occurs while the file
+// is being processed, it's skipped and the function continues with the next one until all files are
+// processed. Nothing is done with those files for which errors occurred, they'll be skipped until either
+// corrected or deleted.
+func (u *Usecases) ProcessManifestFiles(manifestDir string) error {
+
+	manifestFiles, err := os.ReadDir(manifestDir)
+	if err != nil {
+		return fmt.Errorf("Failed to read manifest files from dir %q: %w", manifestDir, err)
+	}
+
+	configs, err := u.ContextBox.GetAllConfigs()
+	if err != nil {
+		return fmt.Errorf("Failed to retrieve manifests details from context-box: %w", err)
+	}
+
+	log.Debug().Msgf("%d configs present in database | %d configs in %v", len(configs), len(manifestFiles), manifestDir)
+
+	manifestProcessingResultsChan := make(chan *ManifestProcessingResult, len(manifestFiles))
+
+	waitGroup := sync.WaitGroup{}
+
+	for _, manifestFile := range manifestFiles {
+		waitGroup.Add(1)
+
+		// Process each of the files concurrently in a separate go-routine skipping over files for which
+		// an error occurs.
+		// By processing, we mean unmarshalling the claudie manifest
+		go func(manifestFile fs.DirEntry) {
+			defer waitGroup.Done()
+
+			manifestFilepath := filepath.Join(manifestDir, manifestFile.Name())
+
+			var (
+				rawManifestData []byte
+				manifest        manifest.Manifest
+				processingError error
+			)
+
+			defer func() {
+				manifestProcessingResultsChan <- &ManifestProcessingResult{
+					manifestName:     manifest.Name,
+					rawManifestData:  rawManifestData,
+					manifestFilepath: manifestFilepath,
+					processingError:  processingError,
+				}
+			}()
+
+			if rawManifestData, processingError = os.ReadFile(manifestFilepath); err != nil {
```

Comment:
`err != nil` the err variable is not one we are after here. I think you meant `processingError != nil`